mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -151,4 +153,77 @@ class StringIndexerModel private[ml] (
     val copied = new StringIndexerModel(uid, labels)
     copyValues(copied, extra)
   }
+
+  /**
+   * Return a model to perform the inverse transformation.
+   * Note: By default we keep the original columns during this transformation, so the inverse
+   * should only be used on new columns such as predicted labels.
+   */
+  def invert(inputCol: String, outputCol: String): StringIndexerInverseTransformer = {
+    val labelsCol: String = $(this.outputCol)
+    new StringIndexerInverseTransformer(labelsCol)
+      .setInputCol(inputCol)
+      .setOutputCol(outputCol)
+  }
+}
+
+/**
+ * :: Experimental ::
+ * Transform a provided column back to the original input types using the metadata on the
+ * labelsCol. Note: By default we keep the original columns during this transformation,
+ * so the inverse should only be used on new columns such as predicted labels.
+ */
+@Experimental
+class StringIndexerInverseTransformer private[ml] (
+  override val uid: String,
+  val labelsCol: String) extends Transformer
```

Comment:
Cool, let me know if you'd like any help with 7424.

On Wed, Jul 29, 2015 at 3:53 PM, jkbradley <notifications@github.com> wrote:

> In mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala
> <https://github.com/apache/spark/pull/6339#discussion_r35821343>:
>
> > +    new StringIndexerInverseTransformer(labelsCol)
> > +      .setInputCol(inputCol)
> > +      .setOutputCol(outputCol)
> > +  }
> > +}
> > +
> > +/**
> > + * :: Experimental ::
> > + * Transform a provided column back to the original input types using the metadata on the
> > + * labelsCol. Note: By default we keep the original columns during this transformation,
> > + * so the inverse should only be used on new columns such as predicted labels.
> > + */
> > +@Experimental
> > +class StringIndexerInverseTransformer private[ml] (
> > +  override val uid: String,
> > +  val labelsCol: String) extends Transformer
>
> There is: SPARK-7424
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/6339/files#r35821343>.
>



-- 
Cell : 425-233-8271
Twitter: https://twitter.com/holdenkarau
Linked In: https://www.linkedin.com/in/holdenkarau